descartes_utilities/src/ros_conversions.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ *  ros_conversions.cpp
+ *
+ *  Created on: Feb 28, 2016
+ *  Author: Jonathan Meyer
+ */
+
+#include "descartes_utilities/ros_conversions.h"
+#include <algorithm>
+#include "console_bridge/console.h"
+
+static double minTime(const std::vector<double>& a, const std::vector<double>& b, double max_vel)
+{
+  std::vector<double> diff;
+  diff.reserve(a.size());
+
+  // compute joint-wise minimum time required based on relative distance between joint positions
+  // and the maximum allowable joint velocity
+  std::transform(a.begin(), a.end(), b.begin(), std::back_inserter(diff), [max_vel] (double a, double b) {
+    return std::abs(a - b) / max_vel;
+  });
+  // The biggest time across all of the joints is the min time
+  return *std::max_element(diff.begin(), diff.end());
+}
+
+bool
+descartes_utilities::toRosJointPoints(const descartes_core::RobotModel& model,
+                                      const std::vector<descartes_core::TrajectoryPtPtr>& joint_traj,
+                                      double default_joint_vel,
+                                      std::vector<trajectory_msgs::JointTrajectoryPoint>& out)
+{
+  if (default_joint_vel <= 0.0)
+  {
+    logError("%s: Invalid value for default joint velocity. Must be > 0 (radians/second)", __FUNCTION__);
+    return false;
+  }
+
+  ros::Duration from_start (0.0);
+  std::vector<trajectory_msgs::JointTrajectoryPoint> ros_trajectory;
+  ros_trajectory.reserve(joint_traj.size());
+
+  std::vector<double> joint_point;
+  std::vector<double> dummy;
+
+  for (std::size_t i = 0; i < joint_traj.size(); ++i)
+  {
+    if (!joint_traj[i])
+    {
+      logError("%s: Input trajectory contained null pointer at index %lu", __FUNCTION__, i);
```

Comment:
Yeah, so I had to fix this somewhere myself recently, and the issue is that `%zu` apparently is C99, and not necessarily portable across all C++ compilers (if they support it at all). Afaict (but I'd like to be wrong), the only really portable approach is to just cast to something that *does* fit `%lu` (or whatever you want to use).

Alternatively, using C++ stream operators also works.